lib/simplifyExpression/basicsSearch/reduceMultiplicationByZero.js

```diff
@@ -1,25 +1,21 @@
+const {query: q} = require('math-nodes');
+const {rewriteNode} = require('math-rules');
+
 const ChangeTypes = require('../../ChangeTypes');
+const defineRuleString = require('../../util/defineRuleString');
 const Node = require('../../node');
 
+// The math-rules rule for this didn't do the right thing when the '0' was
+// in the middle of the args list.
+const REDUCE_MULTIPLY_BY_ZERO = defineRuleString('#a', '0', {
+  a: node => q.isMul(node) && node.args.some(arg => q.getValue(arg) === 0),
```

Comment:
I wonder if this would be easier to read multi-line (I get it now, but I also know what the rule is supposed to do)